packages/@sfcc-bff/basketapi/src/api/schema/basketResolvers.ts

```diff
@@ -256,6 +256,64 @@ const removeItemFromBasket = async (
     return getBasket(config, context);
 };
 
+const addCouponToBasket = async (
+    couponCode: string,
+    config: Config,
+    context: AppContext,
+) => {
+    const basketId = context.getSessionProperty('basketId');
+    if (!basketId) {
+        return {
+            fault: {
+                type: 'NoBasketCreated',
+                message: 'No Basket has been created yet.',
+            },
+        };
+    }
+
+    const basketClient = await getBasketClient(config, context);
+    return basketClient
+        .addCouponToBasket({
+            parameters: {
+                basketId: basketId,
+            },
+            body: {
+                code: couponCode,
+            },
+        })
+        .catch(e => {
+            logger.error(
+                `Error in submitCouponToBasket() for coupon code: ${couponCode}`,
+            );
+            throw e;
+        });
+};
+
+const removeCouponFromBasket = async (
+    couponItemId: string,
+    config: Config,
+    context: AppContext,
+) => {
+    const basketId = context.getSessionProperty('basketId');
+    if (!basketId) {
+        return {
```

Comment:
This return statement appears to be very similar to the return statement on L226. Should we consider moving this out consider we may utilize the same message in other places?

Reply:
good point, will do it.